scipy/optimize/tests/test__gensa.py

```diff
@@ -0,0 +1,79 @@
+"""
+Unit tests for the Generalized Simulated Annealing global optimizer
+"""
+from scipy.optimize import _gensa
+from scipy.optimize._gensa import GenSARunner
+from scipy.optimize._gensa import GenSARunnerException
+import numpy as np
+from numpy.testing import (assert_equal, TestCase, assert_allclose,
+                           assert_almost_equal, assert_raises)
+
+
+class TestGenSA(TestCase):
+
+    def setUp(self):
+        # Using Rastrigin function for performing tests
+        self.func = lambda x: np.sum(x * x - 10 * np.cos(2 * np.pi * x))\
+                + 10 * np.size(x)
+        self.weirdfunc = lambda x: 1e15
+        self.ld_lowerb = [-5.12] * 2
+        self.ld_upperb = [5.12] * 2
+        self.hd_lowerb = self.ld_lowerb * 5
+        self.hd_upperb = self.ld_upperb * 5
+        self.nbtestvalues = 5000
+        self.defautgr = (self.func, None, self.ld_lowerb, self.ld_upperb)
+
+    def tearDown(self):
+        pass
+
+    def test_low_dim(self):
+        ret = _gensa.gensa(self.func, None, (zip(self.ld_lowerb,
+            self.ld_upperb)))
+        assert_allclose(ret.fun, 0., atol=1e-12)
+
```

Comment:
add test that checks that setting a random seed gives reproducible minimization.